### ISSUE TITLE
use area def raise value different when process himawari datasets

### ISSUE THREAD

**Issue by ceri540** [pytroll/satpy#2323]
**Describe the bug**
I want to convert the projection of himawari 8 into a plane coordinate system, so I use `create_area_def` to achieve projection transformation and area clipping. But in the process, I found that there is a big difference in the band value trend between  use `create_area_def` and `scn.crop`. The trend of values ​​at the same points should be similar, but using create_area_def seems to enhance the B04,B05,B06 band. I'm new to satpy so I'm not sure if I'm doing something wrong.

**To Reproduce**

```python
import os
import numpy as np
from datetime import datetime

from satpy import Scene, find_files_and_readers
from satpy.dataset import DataQuery
from pyresample import create_area_def


year = '2016'
month = '01'
day = '06'
hour = '02' 
ds_dir =  f"./himawari/{year}/{month}/{day}/{hour}"

# load datasets and define modifiers
files = find_files_and_readers(sensor="ahi",
                               start_time=datetime(int(year), int(month), int(day), int(hour), 30),
                               end_time=datetime(int(year), int(month), int(day), int(hour), 30),
                               base_dir=ds_dir,
                               reader='ahi_hsd')
scn = Scene(filenames=files)
scn.load([
    DataQuery(name = 'B01', modifiers = ('sunz_corrected', )),
    DataQuery(name = 'B02', modifiers = ('sunz_corrected', )), 
    DataQuery(name = 'B03', modifiers = ('sunz_corrected', )),
    DataQuery(name = 'B04', modifiers = ('sunz_corrected', )),
    DataQuery(name = 'B05', modifiers = ('sunz_corrected', )),
    DataQuery(name = 'B06', modifiers = ('sunz_corrected', )),
    
    DataQuery(name = 'B01', modifiers = ('sunz_corrected', 'rayleigh_corrected')),
    DataQuery(name = 'B02', modifiers = ('sunz_corrected', 'rayleigh_corrected')), 
    DataQuery(name = 'B03', modifiers = ('sunz_corrected', 'rayleigh_corrected')),
    ])
# clip region by crop and area def
LonLat_bbox = (111.0, 21.13, 116.0, 23.0)
resolution = 0.005
scn_crop = scn.crop(ll_bbox=LonLat_bbox)
scn_crop_resample = scn_crop.resample(scn_crop.max_area(), resampler='native')
area_def = create_area_def(
    'my_area',
    {'proj': 'longlat', 'datum': 'WGS84'},
    area_extent = list(LonLat_bbox),
    resolution = resolution,
    units='degrees'
    )
scn_resample = scn.resample(area_def, resampler='nearest')

```

**Expected behavior**
The trend of values ​​at the same point should be similar.
the trend of B01 ~ B06 by crop
![image](https://user-images.githubusercontent.com/86223697/207318829-c6dab23b-2a5d-4f51-90d0-f7f2a4b2f214.png)

**Actual results**
the B04,B05,B06 band is  enhanced 
the trend of B01 ~ B06 by create_area_def
![image](https://user-images.githubusercontent.com/86223697/207318875-51ac00b2-65ef-4ae9-86a7-2b6f93970948.png)

**Screenshots**

**Environment Info:**
 - OS: [Linux Ubuntu 16.04]
 - Satpy Version: [0.37.1]
 - PyResample Version: [1.25.1]

**Additional context**

**Comment by ceri540** [comment]
> 感谢您提交问题。我想我一般会同意你的断言，我可能会对发生的事情有一些猜测，但首先：你在这里密谋什么？这是单个像素的反射率吗？平均？“之前”和“之后”分别指的是什么？

In fact, what I hope to get is the correction value of Rayleigh scattering obtained by subtracting the reflectance after Rayleigh scattering from the reflectance before Rayleigh scattering, and at the same time realize himawari 8
The satellite data is converted from spherical coordinates to plane coordinates, because in subsequent calculations I need their latitude and longitude grids and spherical coordinates will produce a certain deviation.

**Comment by ceri540** [comment]
> Thanks for submitting the issue!
> 
> Could you complement the code example to generate the plots also?
> 
> I'm a bit suspicious about loading both rayleigh corrected and not corrected same channels in the same scene, I'm wondering if this could be an issue where one is used by satpy wrongly instead of the other? Bands 4 5 and 6 look to be identical, while the ones which also have rayleigh correction look different...

and I've tried loading them separately and it doesn't seem to make a difference
here is code for seperately loading them
```python
ds_dir =  f"/himawari/{year}/{month}/{day}/{hour}"

files = find_files_and_readers(sensor="ahi",
                               start_time=datetime(int(year), int(month), int(day), int(hour), 30),
                               end_time=datetime(int(year), int(month), int(day), int(hour), 30),
                               base_dir=ds_dir,
                               reader='ahi_hsd')
scn = Scene(filenames=files)
new_scn = Scene(filenames=files)
scn.load([
    DataQuery(name = 'B01', modifiers = ('sunz_corrected', 'rayleigh_corrected')),
    DataQuery(name = 'B02', modifiers = ('sunz_corrected', 'rayleigh_corrected')), 
    DataQuery(name = 'B03', modifiers = ('sunz_corrected', 'rayleigh_corrected')),
    ])
new_scn.load([
    DataQuery(name = 'B01', modifiers = ('sunz_corrected', )),
    DataQuery(name = 'B02', modifiers = ('sunz_corrected', )), 
    DataQuery(name = 'B03', modifiers = ('sunz_corrected', )),
    DataQuery(name = 'B04', modifiers = ('sunz_corrected', )),
    DataQuery(name = 'B05', modifiers = ('sunz_corrected', )),
    DataQuery(name = 'B06', modifiers = ('sunz_corrected', )),])

LonLat_bbox = (111.0, 21.13, 116.0, 23.0)
resolution = 0.005
area_def = create_area_def(
    'my_area',
    {'proj': 'longlat', 'datum': 'WGS84'},
    area_extent = list(LonLat_bbox),
    resolution = resolution,
    units='degrees'
    )
scn_resample = scn.resample(area_def, resampler='nearest')
new_scn_resample = new_scn.resample(area_def, resampler='nearest')

ds_after = scn_resample.keys()
ds_before = new_scn_resample.keys()
row = 150
col = 500

B01_after = scn_resample[ds_after[0]].values[row, col]
B02_after = scn_resample[ds_after[1]].values[row, col]
B03_after = scn_resample[ds_after[2]].values[row, col]

B01_before = new_scn_resample[ds_before[0]].values[row, col]
B02_before = new_scn_resample[ds_before[1]].values[row, col]
B03_before = new_scn_resample[ds_before[2]].values[row, col]
B04_before = new_scn_resample[ds_before[3]].values[row, col]
B05_before = new_scn_resample[ds_before[4]].values[row, col]
B06_before = new_scn_resample[ds_before[5]].values[row, col]

y_before = [
    B01_before,
    B02_before,
    B03_before,
    B04_before,
    B05_before,
    B06_before,
]
y_after = [
    B01_after,
    B02_after,
    B03_after,
    B04_before,
    B05_before,
    B06_before,
]
y_rayleigh = [y_before[i]-_ for i, _ in enumerate(y_after)]
x = [1, 2, 3, 4, 5, 6]
plt.figure(dpi=600)
plt.plot(x,y_before,'-bo', label='before')
plt.plot(x,y_after,'-ro',  label='after')
plt.plot(x,y_rayleigh,'-ko',label='rayleigh reflectance')
plt.legend(frameon=False)
plt.xlabel("Band name")
plt.ylabel("Reflectance")
plt.title("reflectance of B01-B06 by using area def and nearest to resample")
plt.show()
```

the result show like the picture
![image](https://user-images.githubusercontent.com/86223697/207338302-06e3b6b4-40bf-4bdd-8270-b33b07638aa0.png)

**Comment by mraspaud** [comment]
Ok, thanks for the precisions. 
The black curves look quite similar between the plots I have to say, but not the other ones indeed.

I can see that the resampling is using an are in longlat projection, which quite different from the native projection of the data that crop preserves. But indeed, the difference should probably not be that big. Just to make sure we are on the same domain in both cases, would you mind posting here an image of eg B01 resampled to your area, and the same channel cropped over that lonlat box? Maybe we can then see what the big difference is?

**Comment by ceri540** [comment]
Sincerely thank you both for your attention to my question. Today, I conducted an experiment on some pixels of the clean ocean atmosphere and found that it conforms to the law. At the same time, after visualizing the raw data from himawari’s official website, I found that the problem does not come from processing but from high-level Anomalies at bright pixels. Sorry I'm not familiar with dealing with this part of data causing this problem.

![image](https://user-images.githubusercontent.com/86223697/207529738-6c4eff6d-db42-4210-b96b-0c9040be2480.png)
![image](https://user-images.githubusercontent.com/86223697/207529762-0148e2eb-2108-4f50-b434-9d0e5700741b.png)
